File: app/src/main/java/com/jawnnypoo/geotune/misc/AnimUtils.java
package com.jawnnypoo.geotune.misc;

import android.animation.Animator;
import android.annotation.TargetApi;
import android.view.View;
import android.view.ViewAnimationUtils;
import android.view.animation.AccelerateInterpolator;

/**
 * Animation stuff
 */
public class AnimUtils {

    private static final AccelerateInterpolator mAccelerateInterpolator = new AccelerateInterpolator();

    @TargetApi(21)
    public static Animator circleReveal(View v, int centerX, int centerY,
                                        float startRadius, float endRadius, int duration) {
        Animator anim = ViewAnimationUtils.createCircularReveal(v, centerX, centerY, startRadius, endRadius);
        anim.setDuration(duration);
        anim.setInterpolator(mAccelerateInterpolator);
        anim.start();
        return anim;
    }
}

File: app/src/main/java/com/jawnnypoo/geotune/util/AnimUtils.kt
package com.jawnnypoo.geotune.util

import android.animation.Animator
import android.annotation.TargetApi
import android.view.View
import android.view.ViewAnimationUtils
import android.view.animation.AccelerateInterpolator

/**
 * Animation stuff
 */
object AnimUtils {

    private val mAccelerateInterpolator = AccelerateInterpolator()

    @TargetApi(21)
    fun circleReveal(v: View, centerX: Int, centerY: Int,
                     startRadius: Float, endRadius: Float, duration: Int): Animator {
        val anim = ViewAnimationUtils.createCircularReveal(v, centerX, centerY, startRadius, endRadius)
        anim.duration = duration.toLong()
        anim.interpolator = mAccelerateInterpolator
        anim.start()
        return anim
    }
}
